### DISCUSSION TITLE
zh/integrations/rockchip-rknn/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#19436] [Docs]
# zh/integrations/rockchip-rknn/

了解如何将YOLO11 模型导出为 RKNN 格式，以便在 Rockchip 平台上高效部署并提高性能。

https://docs.ultralytics.com/zh/integrations/rockchip-rknn/

**Comment by djdll** [comment]
Whether the module supports use yolo p2 p3 p4 p5

**Reply by glenn-jocher** [reply]
@djdll thanks for your question! Currently, RKNN export supports YOLO11 detection models, and we're working to expand support to other architectures. For details on supported models and platforms, please refer to the Rockchip RKNN Export documentation. Let us know if you have specific use cases we should consider for future updates! 🚀

**Comment by UltralyticsAssistant** [comment]
👋 Hello, thank you for your interest in Ultralytics 🚀! The integration of YOLO models with Rockchip platforms is an exciting area of development 🌟. To get started, we recommend exploring the [Rockchip RKNN Integration Guide](https://docs.ultralytics.com/zh/integrations/rockchip-rknn/), which provides a step-by-step walkthrough for exporting YOLO models to the RKNN format.

If you are encountering any issues or have specific questions, please include further details, such as the model you're exporting, the platform environment, and any error messages or logs. If this is a 🐛 Bug Report, kindly provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate more effectively.

For general tips and best practices, make sure to also check out our resources for [custom model training and deployment](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best to connect with other users and our community of experts:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time conversations
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) to learn and share experiences

## Upgrade

If you haven’t already, ensure you’re using the latest version of the `ultralytics` package. Use the following command to upgrade:

```bash
pip install -U ultralytics
```

Verify that your development environment meets all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), including [**Python>=3.8**](https://www.python.org/) and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).

## Environments

To explore and experiment efficiently, you can run YOLO in these verified environments with all dependencies preinstalled:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

The current status of our CI tests is available here:  
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>  
These tests verify that all YOLO modes and tasks are functioning correctly across multiple platforms.

This is an automated response to assist you promptly 😊. An Ultralytics engineer will review and provide further assistance soon!

**Comment by HuangCode-YZ** [comment]
Does it support yolov8?

**Reply by HuangCode-YZ** [reply]
Error during inference：Trying to create tensor with negative dimension -990

**Reply by glenn-jocher** [reply]
@HuangCode-YZ yes, YOLOv8 models are supported for RKNN export. You can export YOLOv8 detection models using the same process shown in our Rockchip RKNN export guide by specifying your YOLOv8 model file (e.g., `yolov8n.pt`).

**Comment by as-jackson** [comment]
We are pleased to confirm that the following ArmSoM boards based on Rockchip RK3576 have been successfully tested with Ultralytics projects (e.g., YOLO11n):

[ArmSoM-Sige5](https://docs.armsom.org/armsom-sige5)
[ArmSoM-CM5](https://docs.armsom.org/armsom-cm5)

Validation Scope:

Model training/inference pipeline

**Comment by WeiIIzz** [comment]
Hello, when I exported the rknn model, I performed int8 quantization, but I had an error like this:
E build: If set do_quantization = True, dataset file need be provided. But it`s value is None!
E export_rknn: RKNN model does not exist, please load build model first!
My export code is:
from ultralytics import YOLO

model = YOLO("/home/ky/dfxw/runs/detect/train3/weights/best.pt")
model.export(format="rknn", name="rk3576", int8=True, data="/home/ky/rknn_yolo_test/data.yaml")
The path pointed to by data is the same as the training dataset. How to solve it

**Reply by WeiIIzz** [reply]
The content of data.yaml is as follows
path: /home/ky/dfxw/split_jiaqian # dataset root dir
train: images/train
val: images/val

names:
  0: 0
  1: 1

**Reply by glenn-jocher** [reply]
@WeiIIzz The error occurs because RKNN's quantization process requires a different dataset format than what you're providing. The RKNN toolkit expects specific calibration data rather than just the YAML file path.

For INT8 quantization with RKNN export, try these steps:

1. Create a directory with some sample images from your dataset (8-32 representative images)
2. Use the direct path to that folder instead:

```python
from ultralytics import YOLO

model = YOLO("/home/ky/dfxw/runs/detect/train3/weights/best.pt")
# Point to folder containing image files, not the yaml
model.export(format="rknn", name="rk3576", int8=True, data="/path/to/calibration/images") 
```

The RKNN quantization process works differently than regular YOLO validation - it needs actual image files for calibration rather than a data.yaml configuration. Let us know if this resolves your issue!

**Reply by WeiIIzz** [reply]
@glenn-jocher  I followed your steps to modify the code as follows. But I still encountered the same problem.
```
from ultralytics import YOLO

model = YOLO("/home/ky/dfxw/runs/detect/train3/weights/best.pt")
model.export(format="rknn", name="rk3576", int8=True, data="/home/ky/rknn_yolo_test/images")
```
Is there only images in the data path?

**Reply by glenn-jocher** [reply]
@WeiIIzz I've looked into this issue more deeply. The error occurs because the current implementation of RKNN export in Ultralytics doesn't fully support INT8 quantization yet, despite accepting the parameter. 

Looking at the exporter code, there's actually a comment that says `# TODO: Add quantization support` and it explicitly sets `do_quantization=False` when building the RKNN model, regardless of whether you specify `int8=True` or not.

You have two options:

1. Use the default export without quantization for now:
```python
model.export(format="rknn", name="rk3576")  # without int8 parameter
```

2. If quantization is essential, you can manually quantize your exported model using the RKNN toolkit directly:
```python
from rknn.api import RKNN

# After exporting non-quantized model with Ultralytics
rknn = RKNN(verbose=True)
rknn.load_rknn("path/to/your_exported_model.rknn")
rknn.config(mean_values=[[0, 0, 0]], std_values=[[255, 255, 255]], target_platform="rk3576")
rknn.build(dataset="path/to/calibration/images", do_quantization=True)
rknn.export_rknn("quantized_model.rknn")
```

We'll add full INT8 quantization support in an upcoming update to the RKNN exporter. Thanks for bringing this to our attention!

**Comment by guojieli** [comment]
he YOLO11 benchmark was run by the Ultralytics team on a Radxa Rock 5B based on the Rockchip RK3588, using the RKNN model format to evaluate speed and accuracy. Could you please clarify whether this was an INT8-quantized model?

**Reply by glenn-jocher** [reply]
Those RKNN benchmarks were **not INT8-quantized**—current `rknn` export builds with quantization disabled (`rknn.build(do_quantization=False)`), so results reflect FP32; see the [Rockchip RKNN export guide](https://docs.ultralytics.com/zh/integrations/rockchip-rknn/) and the [Exporter `export_rknn()` implementation reference](https://docs.ultralytics.com/reference/engine/exporter/).
